<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>36 Árvores filogenéticas | Manual de R para Epidemiologistas</title>
<meta name="author" content="the handbook team">
<meta name="description" content="36.1 Visão Geral Árvores filogenéticas são utilizados para visualizar e descrever a relação e evolução dos organismos com base no sequenciamento de seu código genético. Elas podem ser construídos...">
<meta name="generator" content="bookdown 0.35 with bs4_book()">
<meta property="og:title" content="36 Árvores filogenéticas | Manual de R para Epidemiologistas">
<meta property="og:type" content="book">
<meta property="og:description" content="36.1 Visão Geral Árvores filogenéticas são utilizados para visualizar e descrever a relação e evolução dos organismos com base no sequenciamento de seu código genético. Elas podem ser construídos...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="36 Árvores filogenéticas | Manual de R para Epidemiologistas">
<meta name="twitter:description" content="36.1 Visão Geral Árvores filogenéticas são utilizados para visualizar e descrever a relação e evolução dos organismos com base no sequenciamento de seu código genético. Elas podem ser construídos...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.1/transition.js"></script><script src="libs/bs3compat-0.5.1/tabs.js"></script><script src="libs/bs3compat-0.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.29/datatables.js"></script><link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="libs/tabwid-1.1.3/tabwid.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.10/grViz.js"></script><script src="libs/d3-4.5.0/d3.min.js"></script><script src="libs/sankey-1/sankey.js"></script><script src="libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><script src="libs/plotly-binding-4.10.2/plotly.js"></script><script src="libs/typedarray-0.1/typedarray.min.js"></script><link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script><link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="libs/vis-9.1.0/vis-network.min.js"></script><script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script><link href="libs/font-awesome-4.7.0/css/font-awesome.min.css" rel="stylesheet">
<script src="libs/plotly-basic-2.11.1/plotly-basic-2.11.1.min.js"></script><script src="libs/plotly-cartesian-2.11.1/plotly-cartesian-2.11.1.min.js"></script><script src="libs/proj4js-2.3.15/proj4.js"></script><link href="libs/highcharts-9.3.1/css/motion.css" rel="stylesheet">
<script src="libs/highcharts-9.3.1/highcharts.js"></script><script src="libs/highcharts-9.3.1/highcharts-3d.js"></script><script src="libs/highcharts-9.3.1/highcharts-more.js"></script><script src="libs/highcharts-9.3.1/modules/stock.js"></script><script src="libs/highcharts-9.3.1/modules/map.js"></script><script src="libs/highcharts-9.3.1/modules/data.js"></script><script src="libs/highcharts-9.3.1/modules/exporting.js"></script><script src="libs/highcharts-9.3.1/modules/offline-exporting.js"></script><script src="libs/highcharts-9.3.1/modules/drilldown.js"></script><script src="libs/highcharts-9.3.1/modules/item-series.js"></script><script src="libs/highcharts-9.3.1/modules/overlapping-datalabels.js"></script><script src="libs/highcharts-9.3.1/modules/annotations.js"></script><script src="libs/highcharts-9.3.1/modules/export-data.js"></script><script src="libs/highcharts-9.3.1/modules/funnel.js"></script><script src="libs/highcharts-9.3.1/modules/heatmap.js"></script><script src="libs/highcharts-9.3.1/modules/treemap.js"></script><script src="libs/highcharts-9.3.1/modules/sankey.js"></script><script src="libs/highcharts-9.3.1/modules/dependency-wheel.js"></script><script src="libs/highcharts-9.3.1/modules/organization.js"></script><script src="libs/highcharts-9.3.1/modules/solid-gauge.js"></script><script src="libs/highcharts-9.3.1/modules/streamgraph.js"></script><script src="libs/highcharts-9.3.1/modules/sunburst.js"></script><script src="libs/highcharts-9.3.1/modules/vector.js"></script><script src="libs/highcharts-9.3.1/modules/wordcloud.js"></script><script src="libs/highcharts-9.3.1/modules/xrange.js"></script><script src="libs/highcharts-9.3.1/modules/tilemap.js"></script><script src="libs/highcharts-9.3.1/modules/venn.js"></script><script src="libs/highcharts-9.3.1/modules/gantt.js"></script><script src="libs/highcharts-9.3.1/modules/timeline.js"></script><script src="libs/highcharts-9.3.1/modules/parallel-coordinates.js"></script><script src="libs/highcharts-9.3.1/modules/bullet.js"></script><script src="libs/highcharts-9.3.1/modules/coloraxis.js"></script><script src="libs/highcharts-9.3.1/modules/dumbbell.js"></script><script src="libs/highcharts-9.3.1/modules/lollipop.js"></script><script src="libs/highcharts-9.3.1/modules/series-label.js"></script><script src="libs/highcharts-9.3.1/plugins/motion.js"></script><script src="libs/highcharts-9.3.1/custom/reset.js"></script><script src="libs/highcharts-9.3.1/modules/boost.js"></script><script src="libs/highchart-binding-0.9.4/highchart.js"></script><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-QXDW878QLX"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-QXDW878QLX');
    </script>
</head>
<body>
<div class="alert alert-info alert-dismissible">
      <a href="#" class="close" data-dismiss="alert" aria-label="close">×</a>
      <strong>Você precisa de ajuda para aprender R?</strong> Inscreva-se no curso de <a href="https://www.appliedepi.org/live/" class="alert-link">introdução ao R</a> da Applied Epi, experimente nossos <a href="https://www.appliedepi.org/tutorial/" class="alert-link">tutoriais gratuitos sobre o R</a>, publique em nosso <a href="https://community.appliedepi.org/" class="alert-link">fórum de perguntas e respostas</a>, ou solicite nosso <a href="mailto:contact@appliedepi.org" class="alert-link">suporte ao R</a>.
    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style_bs4.css">
<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Manual de R para Epidemiologistas</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"></a></li>
<li class="book-part">Sobre este livro</li>
<li><a class="" href="editorial-style.html"><span class="header-section-number">1</span> Notas editoriais e técnicas</a></li>
<li><a class="" href="data-used.html"><span class="header-section-number">2</span> Baixe o livro e os dados</a></li>
<li class="book-part">Básico</li>
<li><a class="" href="basics.html"><span class="header-section-number">3</span> Introdução ao R</a></li>
<li><a class="" href="transition-to-R.html"><span class="header-section-number">4</span> Transição para o R</a></li>
<li><a class="" href="packages-suggested.html"><span class="header-section-number">5</span> Pacotes sugeridos</a></li>
<li><a class="" href="r-projects.html"><span class="header-section-number">6</span> R projects</a></li>
<li><a class="" href="importing.html"><span class="header-section-number">7</span> Importar e exportar</a></li>
<li class="book-part">Gerenciamento dos Dados</li>
<li><a class="" href="cleaning.html"><span class="header-section-number">8</span> Limpeza de dados e principais funções</a></li>
<li><a class="" href="dates.html"><span class="header-section-number">9</span> Trabalhando com datas</a></li>
<li><a class="" href="characters-strings.html"><span class="header-section-number">10</span> Caracteres e strings</a></li>
<li><a class="" href="factors.html"><span class="header-section-number">11</span> Fatores</a></li>
<li><a class="" href="pivoting.html"><span class="header-section-number">12</span> Pivoteando dados</a></li>
<li><a class="" href="grouping.html"><span class="header-section-number">13</span> Agrupando dados</a></li>
<li><a class="" href="joining-matching.html"><span class="header-section-number">14</span> Juntando dados (Joins)</a></li>
<li><a class="" href="deduplication.html"><span class="header-section-number">15</span> Eliminação de duplicidades</a></li>
<li><a class="" href="iteration.html"><span class="header-section-number">16</span> Iterações, loops e listas</a></li>
<li class="book-part">Análise dos dados</li>
<li><a class="" href="tables-descriptive.html"><span class="header-section-number">17</span> Tabelas descritivas</a></li>
<li><a class="" href="stat-tests.html"><span class="header-section-number">18</span> Testes estatísticos simples</a></li>
<li><a class="" href="regression.html"><span class="header-section-number">19</span> Regressão simples e múltipla</a></li>
<li><a class="" href="missing-data.html"><span class="header-section-number">20</span> Campos em branco/faltantes</a></li>
<li><a class="" href="standardization.html"><span class="header-section-number">21</span> Normalização de taxas</a></li>
<li><a class="" href="moving-average.html"><span class="header-section-number">22</span> Médias móveis</a></li>
<li><a class="" href="epidemic-models.html"><span class="header-section-number">23</span> Modelagem de epidemias</a></li>
<li><a class="" href="contact-tracing.html"><span class="header-section-number">24</span> Rastreamento de contatos</a></li>
<li><a class="" href="survey-analysis.html"><span class="header-section-number">25</span> Analises de pesquisa de questionários (survey)</a></li>
<li><a class="" href="survival-analysis.html"><span class="header-section-number">26</span> Análise de sobrevivência</a></li>
<li class="book-part">Visualização de Dados</li>
<li><a class="" href="tables-presentation.html"><span class="header-section-number">27</span> Tabelas para apresentação</a></li>
<li><a class="" href="ggplot-basics.html"><span class="header-section-number">28</span> O básico do ggplot</a></li>
<li><a class="" href="ggplot-tips.html"><span class="header-section-number">29</span> Dicas do ggplot</a></li>
<li><a class="" href="epicurves.html"><span class="header-section-number">30</span> Curvas epidêmicas</a></li>
<li><a class="" href="age-pyramid.html"><span class="header-section-number">31</span> Pirâmides demográficas e escalas Likert</a></li>
<li><a class="" href="heatmaps.html"><span class="header-section-number">32</span> Gráficos de calor</a></li>
<li><a class="" href="diagrams.html"><span class="header-section-number">33</span> Diagramas e gráficos</a></li>
<li><a class="" href="combination-analysis.html"><span class="header-section-number">34</span> Análise de Combinações</a></li>
<li><a class="" href="transmission-chains.html"><span class="header-section-number">35</span> Cadeias de Transmissão</a></li>
<li><a class="active" href="phylogenetic-trees.html"><span class="header-section-number">36</span> Árvores filogenéticas</a></li>
<li><a class="" href="interactive-plots.html"><span class="header-section-number">37</span> Gráficos interativos</a></li>
<li class="book-part">Relatórios e dashboards</li>
<li><a class="" href="rmarkdown.html"><span class="header-section-number">38</span> Relatórios com R Markdown</a></li>
<li><a class="" href="reportfactory.html"><span class="header-section-number">39</span> Organização de relatórios de rotina</a></li>
<li><a class="" href="flexdashboard.html"><span class="header-section-number">40</span> Painéis (Dashboards) com R Markdown</a></li>
<li><a class="" href="shiny-basics.html"><span class="header-section-number">41</span> Painéis com Shiny</a></li>
<li class="book-part">Miscelânea</li>
<li><a class="" href="writing-functions.html"><span class="header-section-number">42</span> Escrevendo funções</a></li>
<li><a class="" href="directories.html"><span class="header-section-number">43</span> Interações de diretório</a></li>
<li><a class="" href="collaboration.html"><span class="header-section-number">44</span> Controle de versão e colaboração com Git e Github</a></li>
<li><a class="" href="errors.html"><span class="header-section-number">45</span> Erros comuns</a></li>
<li><a class="" href="help.html"><span class="header-section-number">46</span> Conseguindo ajuda</a></li>
<li><a class="" href="network-drives.html"><span class="header-section-number">47</span> R em drives (pastas) na rede</a></li>
<li><a class="" href="data-table.html"><span class="header-section-number">48</span> Data Table</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="phylogenetic-trees" class="section level1" number="36">
<h1>
<span class="header-section-number">36</span> Árvores filogenéticas<a class="anchor" aria-label="anchor" href="#phylogenetic-trees"><i class="fas fa-link"></i></a>
</h1>
<!-- ======================================================= -->
<div id="visão-geral-5" class="section level2" number="36.1">
<h2>
<span class="header-section-number">36.1</span> Visão Geral<a class="anchor" aria-label="anchor" href="#vis%C3%A3o-geral-5"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Árvores filogenéticas</strong> são utilizados para visualizar e descrever a relação e evolução dos organismos com base no sequenciamento de seu código genético.</p>
<p>Elas podem ser construídos a partir de seqüências genéticas usando métodos baseados na distância (como o método de união de vizinhos) ou métodos baseados em caracteres (como o método de máxima verossimilhança e o método Baysiano de Monte Carlo via Cadeias de Markov ). O sequenciamento de próxima geração (NGS) tornou-se mais acessível e está se tornando mais amplamente utilizado na saúde pública para descrever patógenos causadores de doenças infecciosas. Os dispositivos portáteis de seqüenciamento diminuem o tempo de retorno e mantêm promessas de disponibilizar dados para o apoio à investigação de surtos em tempo real. Os dados da NGS podem ser usados para identificar a origem ou fonte de uma variante de surto e sua propagação, bem como determinar a presença de genes de resistência antimicrobiana. Para visualizar a relação genética entre as amostras, é construída uma árvore filogenética.</p>
<p>Nesta página aprenderemos como utilizar o pacote <strong>ggtree</strong>, que permite a visualização combinada de árvores filogenéticas com dados adicionais de amostras sob a forma de um <em>data frame</em>. Isto nos permitirá observar padrões e melhorar a compreensão da dinâmica do surto.</p>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_overview_graph-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="preparação-28" class="section level2" number="36.2">
<h2>
<span class="header-section-number">36.2</span> Preparação<a class="anchor" aria-label="anchor" href="#prepara%C3%A7%C3%A3o-28"><i class="fas fa-link"></i></a>
</h2>
<div id="carregar-pacotes-11" class="section level3 unnumbered">
<h3>Carregar pacotes<a class="anchor" aria-label="anchor" href="#carregar-pacotes-11"><i class="fas fa-link"></i></a>
</h3>
<p>Este trecho de código mostra o carregamento dos pacotes necessárias. Neste manual, enfatizamos <code><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load()</a></code> de <strong>pacman</strong>, que instala o pacote se necessário <em>e</em> o carrega para utilização. Você também pode carregar os pacotes instalados com <code><a href="https://rdrr.io/r/base/library.html">library()</a></code> do R <strong>base</strong>. Veja a página em <a href="basics.html#basics">Introdução ao R</a> para mais informações sobre os pacotes R.</p>
<div class="sourceCode" id="cb1567"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span></span>
<span>  <span class="va">rio</span>,             <span class="co"># importar/exportar</span></span>
<span>  <span class="va">here</span>,            <span class="co"># caminhos relativos dos arquivos</span></span>
<span>  <span class="va">tidyverse</span>,       <span class="co"># manipulações e visualizações gerais de dados</span></span>
<span>  <span class="va">ape</span>,             <span class="co"># para importar e exportar dados de árvores filogenéticas</span></span>
<span>  <span class="va">ggtree</span>,          <span class="co"># para visualizar dados de árvores filogenéticas </span></span>
<span>  <span class="va">treeio</span>,          <span class="co"># para visualizar dados de árvores filogenéticas </span></span>
<span>  <span class="va">ggnewscale</span><span class="op">)</span>      <span class="co"># adicionar camadas de esquema de cores </span></span></code></pre></div>
</div>
<div id="importar-dados-4" class="section level3 unnumbered">
<h3>Importar dados<a class="anchor" aria-label="anchor" href="#importar-dados-4"><i class="fas fa-link"></i></a>
</h3>
<p>Os dados para esta página podem ser baixados com as instruções na página <a href="data-used.html#data-used">Baixar manual e dados</a>.</p>
<p>Há vários formatos diferentes nos quais uma árvore filogenética pode ser armazenada (por exemplo, Newick, NEXUS, Phylip). Um muito comum é o formato de arquivo Newick (.nwk), que é o padrão para representar árvores em formato legível por computador. Isto significa que uma árvore inteira pode ser expressa em formato de string como “((t2:0.04,t1:0.34):0.89,(t5:0.37,(t4:0.03,t3:0.67):0.9):0.59);”, listando todos os nós e ápices e a relação (comprimento do ramo) entre si.</p>
<p>Nota: É importante entender que o arquivo de árvore filogenética em si não contém dados de sequenciamento, mas é meramente o resultado das distâncias genéticas entre as seqüências. Portanto, não podemos extrair dados de seqüenciamento de um arquivo de árvore.</p>
<p>Primeiro, utilizamos a função <code>read.tree()</code> do pacote <strong>ape</strong> para importar um arquivo de árvore filogenética Newick em formato .txt, e armazená-lo em uma lista de objetos da classe “phylo”. Se necessário, utilize a função <code><a href="https://here.r-lib.org//reference/here.html">here()</a></code> do pacote <strong>here</strong> para especificar o caminho relativo do arquivo.</p>
<p>Nota: Neste caso, a árvore newick é salva como um arquivo .txt para facilitar o manuseio e o download do Github.</p>
<div class="sourceCode" id="cb1568"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="st">"Shigella_tree.txt"</span><span class="op">)</span></span></code></pre></div>
<p>Inspecionamos nosso objeto árvore e vemos que ele contém 299 ápices/pontas (ou amostras) e 236 nós.</p>
<div class="sourceCode" id="cb1569"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tree</span></span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 299 tips and 236 internal nodes.
## 
## Tip labels:
##   SRR5006072, SRR4192106, S18BD07865, S18BD00489, S17BD08906, S17BD05939, ...
## Node labels:
##   17, 29, 100, 67, 100, 100, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>Segundo, importamos uma tabela armazenada como um arquivo .csv com informações adicionais para cada amostra sequenciada, como sexo, país de origem e atributos de resistência antimicrobiana, utilizando a função <code><a href="https://rdrr.io/pkg/rio/man/import.html">import()</a></code> do pacote <strong>rio</strong>:</p>
<div class="sourceCode" id="cb1571"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rio/man/import.html">import</a></span><span class="op">(</span><span class="st">"sample_data_Shigella_tree.csv"</span><span class="op">)</span></span></code></pre></div>
<p>Abaixo estão as primeiras 50 linhas de observação dos dados:</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-5e8096cae725f71bfe8f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5e8096cae725f71bfe8f">{"x":{"filter":"none","vertical":false,"data":[["S17BD05944","S15BD07413","S18BD07247","S19BD07384","S18BD07338","S18BD02657","S19BD06884","S16BD02574","S16BD04069","S17BD08170","S15BD07250","S17BD07935","S17BD00384","S19BD06034","S17BD08948","S17BD04542","S17BD07399","S17BD08233","S19BD07381","S17BD07961","S18BD09176","S19BD07393","S18BD02580","S17BD02526","S17BD06833","S18BD01024","S18BD03411","S18BD09515","S17BD07332","S18BD06598","S17BD02164","S19BD06386","S19BD06283","S19BD06285","S19BD05221","S18BD06749","S17BD02088","S17BD07692","S14BD03573","S17BD01135","S19BD06689","S15BD06969","S18BD01106","S17BD00525","S19BD07036","S18BD06401","S18BD06486","S17BD06426","S17BD05916","S18BD07708"],["Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei","Sonnei"],["Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium","Belgium"],["Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe","Europe"],[null,"Georgia",null,"Morocco",null,null,null,"Thailand",null,null,null,null,null,null,null,"Iran",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Morocco","France","Turkey","Egypt","Tunisia","Congo",null,null,null,null,null,null,null,null,null,null,"Egypt",null,null],["2017","2015","2018","2019","2018","2018","2019","2016","2016","2017","2015","2017","2017","2019","2017","2017","2017","2017","2019","2017","2018","2019","2018","2017","2017","2018","2018","2018","2017","2018","2017","2019","2019","2019","2019","2018","2017","2017","2014","2017","2019","2015","2018","2017","2019","2018","2018","2017","2017","2018"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL","NRC BEL"],[null,"Male","Female","Female","Male","Male","Male","Male","Male","Male","Female",null,"Male","Male","Female","Male","Male","Female","Female","Male","Male","Male","Male","Female","Male","Female","Female","Male","Female","Male","Female","Female","Female","Female","Female","Female","Male","Male",null,"female","Male","Female","Male","Male","Male","Male","Male","Male","Male","Female"],[null,null,null,null,null,"gyrA D87Y","gyrA D87Y","gyrA D87Y","gyrA D87Y",null,null,"gyrA D87Y",null,null,"gyrA D87Y",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA S83L",null,null,null,null,null,null,null,null,null,null,"gyrA D87Y","gyrA D87Y",null,null,null,"gyrA D87Y",null,null,null,null,null,null,null],[null,null,null,null,null,"mphA","mphA",null,null,null,null,"mphA",null,null,"mphA",null,null,null,null,null,null,null,null,null,null,"mphA","mphA",null,null,null,null,null,null,null,"mphA/ermB",null,null,"mphA",null,null,null,null,"mphA",null,null,null,null,null,null,null],[8,0,16,8,8,64,64,0,0,64,0,64,8,8,64,8,8,8,8,8,8,16,8,8,8,64,64,8,8,8,8,8,8,2,64,8,8,64,0,8,8,0,64,8,4,8,8,8,8,8],[0.25,0,0.03,0.02,0.02,0.5,0.5,0,0,8,0,0.5,0.02,0.02,0.5,0.25,0.02,0.02,0.02,0.02,0.02,0.12,0.02,0.02,0.02,0.5,0.25,0.03,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,0.02,1,0,0.02,0.02,0,0.5,0.02,0.02,4,0.02,0.02,0.25,0.02]],"container":"<table class=\"white-space: nowrap\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>serotype<\/th>\n      <th>Country<\/th>\n      <th>Continent<\/th>\n      <th>Travel_history<\/th>\n      <th>Year<\/th>\n      <th>Belgium<\/th>\n      <th>Source<\/th>\n      <th>Gender<\/th>\n      <th>gyrA_mutations<\/th>\n      <th>macrolide_resistance_genes<\/th>\n      <th>MIC_AZM<\/th>\n      <th>MIC_CIP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[11,12]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="limpar-e-inspecionar" class="section level3 unnumbered">
<h3>Limpar e inspecionar<a class="anchor" aria-label="anchor" href="#limpar-e-inspecionar"><i class="fas fa-link"></i></a>
</h3>
<p>Nós limpamos e inspecionamos nossos dados: A fim de atribuir os dados da amostra correta à árvore filogenética, os valores na coluna <code>Sample_ID</code> no data frame <code>sample_data</code> precisam corresponder aos valores <code>tip.labels</code> do arquivo <code>tree</code>:</p>
<p>Verificamos a formatação das <code>tip.labels</code> no arquivo <code>tree</code> observando as primeiras 6 entradas utilizando com <code><a href="https://rdrr.io/r/utils/head.html">head()</a></code> do R <strong>base</strong>.</p>
<div class="sourceCode" id="cb1572"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span> </span></code></pre></div>
<pre><code>## [1] "SRR5006072" "SRR4192106" "S18BD07865" "S18BD00489" "S17BD08906" "S17BD05939"</code></pre>
<p>Também nos certificamos de que a primeira coluna em nosso data frame “Sample_data” seja “Sample_ID”. Olhamos os nomes das colunas de nosso data frame utilizando <code><a href="https://rdrr.io/r/base/colnames.html">colnames()</a></code> dm R <strong>base</strong>.</p>
<div class="sourceCode" id="cb1574"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">)</span>   </span></code></pre></div>
<pre><code>##  [1] "Sample_ID"                  "serotype"                  
##  [3] "Country"                    "Continent"                 
##  [5] "Travel_history"             "Year"                      
##  [7] "Belgium"                    "Source"                    
##  [9] "Gender"                     "gyrA_mutations"            
## [11] "macrolide_resistance_genes" "MIC_AZM"                   
## [13] "MIC_CIP"</code></pre>
<p>Olhamos para os <code>Sample_IDs</code> no data frame para ter certeza de que a formatação é a mesma do que na <code>tip.label</code> (por exemplo, todas as letras são maiúsculas, sem sublinhados extras `_’ entre letras e números, etc.)</p>
<div class="sourceCode" id="cb1576"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">)</span> <span class="co"># novamente inspecionamos os primeiros 6 usando head()</span></span></code></pre></div>
<pre><code>## [1] "S17BD05944" "S15BD07413" "S18BD07247" "S19BD07384" "S18BD07338" "S18BD02657"</code></pre>
<p>Também podemos comparar se todas as amostras estão presentes no arquivo “tree” e vice-versa, gerando um vetor lógico de VERDADEIRO ou FALSO onde elas coincidem ou não. Estas não são impressas aqui, por simplicidade.</p>
<div class="sourceCode" id="cb1578"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span></span>
<span></span>
<span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></span></code></pre></div>
<p>Podemos usar estes vetores para mostrar qualquer identificação de amostra que não esteja na árvore (não há nenhuma).</p>
<div class="sourceCode" id="cb1579"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">[</span><span class="op">!</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span><span class="op">]</span></span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>Após a inspeção, podemos ver que o formato do <code>Sample_ID</code> no data frame corresponde ao formato dos nomes das amostras nas <code>tip.labels</code>. Estas não precisam ser classificadas na mesma ordem para serem combinadas.</p>
<p>Estamos prontos para ir!</p>
<!-- ======================================================= -->
</div>
</div>
<div id="visualização-simples-da-árvore" class="section level2" number="36.3">
<h2>
<span class="header-section-number">36.3</span> Visualização simples da árvore<a class="anchor" aria-label="anchor" href="#visualiza%C3%A7%C3%A3o-simples-da-%C3%A1rvore"><i class="fas fa-link"></i></a>
</h2>
<div id="diferente-layouts-para-a-árvore" class="section level3 unnumbered">
<h3>Diferente layouts para a árvore<a class="anchor" aria-label="anchor" href="#diferente-layouts-para-a-%C3%A1rvore"><i class="fas fa-link"></i></a>
</h3>
<p><strong>ggtree</strong>* oferece muitos formatos de layout diferentes e alguns podem ser mais adequados para seu propósito específico do que outros. Abaixo estão algumas demonstrações. Para outras opções, veja este <a href="http://yulab-smu.top/treedata-book/chapter4.html">livro online</a>.</p>
<p>Aqui estão alguns exemplos de layouts de árvores:</p>
<div class="sourceCode" id="cb1581"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>                                            <span class="co"># árvore linear simples</span></span>
<span><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,  branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>                   <span class="co"># árvore linear simples com todas as pontas alinhadas</span></span>
<span><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span>                         <span class="co"># árvore circular simples</span></span>
<span><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="co"># árvore circular simples com todas as pontas alinhadas</span></span></code></pre></div>
<p><img src="_main_files/figure-html/phylogenetic_trees_example_formats-1.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-2.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-3.png" width="50%"><img src="_main_files/figure-html/phylogenetic_trees_example_formats-4.png" width="50%"></p>
</div>
<div id="árvore-simples-com-dados-de-amostras" class="section level3 unnumbered">
<h3>Árvore simples com dados de amostras<a class="anchor" aria-label="anchor" href="#%C3%A1rvore-simples-com-dados-de-amostras"><i class="fas fa-link"></i></a>
</h3>
<p>O operador <strong>%&lt;+%</strong>* é utilizado para conectar o data frame <code>sample_data</code> ao arquivo <code>tree</code>.
A anotação mais fácil de sua árvore é a adição dos nomes das amostras nas pontas, assim como a coloração dos pontos da ponta e, se desejado, dos ramos:</p>
<p>Aqui está um exemplo de uma árvore circular:</p>
<div class="sourceCode" id="cb1582"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, layout <span class="op">=</span> <span class="st">"circular"</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span> <span class="co"># %&lt;+% unifica o datarframe com os dados de amostras à arvore</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="op">(</span><span class="va">Belgium</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># colore os ramos de acordo com a variável no seu dataframe c</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Sample Origin"</span>,                      <span class="co"># nome do seu esquema de cotes (irá aparecer assim na legenda) </span></span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Yes"</span>, <span class="st">"No"</span><span class="op">)</span>,                     <span class="co"># as diferentes opções na sua variável</span></span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NRCSS Belgium"</span>, <span class="st">"Other"</span><span class="op">)</span>,        <span class="co"># como você quer que as diferentes opções sejam nomeadas na sua legenda, permite formatação </span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span><span class="op">)</span>,                  <span class="co"># a cor que você atribuir à variável</span></span>
<span>    na.value <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span><span class="op">+</span>                        <span class="co"># colore valores NA como preto também</span></span>
<span>  <span class="fu">new_scale_color</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>                             <span class="co"># permite adicionar um novo esquema de cores para uma nova variável</span></span>
<span>    <span class="fu">geom_tippoint</span><span class="op">(</span></span>
<span>      mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">Continent</span><span class="op">)</span>,          <span class="co"># cor das pontas por continente. Você pode mudar o formato adicionando "shape = "</span></span>
<span>      size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span>                               <span class="co"># define o tamanho do ponto na ponta </span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Continent"</span>,                    <span class="co"># nome do seu esquema de cotes (irá aparecer assim na legenda) </span></span>
<span>    palette <span class="op">=</span> <span class="st">"Set1"</span>,                      <span class="co">#  escolhemos uma paleta de cores do pacote Brewer </span></span>
<span>    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span><span class="op">+</span>                    <span class="co"># para valores NA escolhemos o cinza </span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>                             <span class="co"># adiciona o nome da amostra para a ponta do ramo </span></span>
<span>    color <span class="op">=</span> <span class="st">'black'</span>,                       <span class="co"># (adiciona quantas linhas de texto desejar, mas talvez precise ajustar o valor de partida para coloca-los proximos uns aos outros. </span></span>
<span>    offset <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    geom <span class="op">=</span> <span class="st">"text"</span>,</span>
<span>    <span class="co">#align = TRUE</span></span>
<span>    <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Árvore filogenética de Shigella sonnei"</span><span class="op">)</span><span class="op">+</span>       <span class="co"># título do seu gráfico</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, <span class="co"># remove título de eixo x</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, <span class="co"># remove título de eixo y</span></span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>    <span class="co"># define o tamanho da fonte e formata a o título da legenda</span></span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,   </span>
<span>    legend.text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>       <span class="co"># define o tamanho da fonte e formata o texto da legenda</span></span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,  </span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>      <span class="co"># define o tamanho da fonte e formata o título do gráfico</span></span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,  </span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,     <span class="co"># define a posição da legenda</span></span>
<span>    legend.box <span class="op">=</span> <span class="st">"vertical"</span>,        <span class="co"># define o posicionamento da legenda</span></span>
<span>    legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>   </span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_adding_sampledata-1.png" width="672" style="display: block; margin: auto;"></div>
<p>Você pode exportar seu gráfico de árvore com <code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave()</a></code> como qualquer outro objeto ggplot’. Escrito desta forma, <code><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave()</a></code> salva a última imagem produzida para o caminho do arquivo que você especificar. Lembre-se de que você pode utilizar <code><a href="https://here.r-lib.org//reference/here.html">here()</a></code> e caminhos de arquivo relativos para salvar facilmente em subpastas, etc.</p>
<div class="sourceCode" id="cb1583"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"example_tree_circular_1.png"</span>, width <span class="op">=</span> <span class="fl">12</span>, height <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<!-- ======================================================= -->
</div>
</div>
<div id="manipulação-da-árvore" class="section level2" number="36.4">
<h2>
<span class="header-section-number">36.4</span> Manipulação da árvore<a class="anchor" aria-label="anchor" href="#manipula%C3%A7%C3%A3o-da-%C3%A1rvore"><i class="fas fa-link"></i></a>
</h2>
<p>Às vezes você pode ter uma árvore filogenética muito grande e só está interessado em uma parte da árvore. Por exemplo, se você produziu uma árvore incluindo amostras históricas ou internacionais para obter uma grande visão geral de onde seu conjunto de dados pode se encaixar no quadro geral. Mas então, para olhar mais de perto seus dados, você quer inspecionar apenas aquela parte da árvore maior.</p>
<p>Como o arquivo filogenético da árvore é apenas a saída da análise sequencial de dados, não podemos manipular a ordem dos nós e ramos no próprio arquivo. Estes já foram determinados em análises anteriores a partir dos dados brutos do NGS. No entanto, somos capazes de fazer zoom em partes, esconder partes e até mesmo subdividir parte da árvore.</p>
<div id="aumentar-o-zoom" class="section level3 unnumbered">
<h3>Aumentar o zoom<a class="anchor" aria-label="anchor" href="#aumentar-o-zoom"><i class="fas fa-link"></i></a>
</h3>
<p>Se você não quiser “cortar” sua árvore, mas apenas inspecionar parte dela mais de perto, você pode ampliar para ver uma parte específica.</p>
<p>Primeiro, plotamos a árvore inteira em formato linear e adicionamos etiquetas numéricas a cada nó da árvore.</p>
<div class="sourceCode" id="cb1584"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>,<span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> <span class="co"># adicionamos rótulos às pontas de todos os ramos como nome da amostra no arquivo </span></span>
<span>  <span class="fu">geom_text2</span><span class="op">(</span></span>
<span>    mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>,</span>
<span>                  label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"darkred"</span>,</span>
<span>    hjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>                            <span class="co"># adiciona rótulos em todos os nós</span></span>
<span></span>
<span><span class="va">p</span>  <span class="co"># printa</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Para aumentar o zoom para um ramo em particular (à direita), utilize <code>viewClade()</code> no objeto ggtree <code>p</code> e forneça o número do nó para obter uma visão mais detalhada:</p>
<div class="sourceCode" id="cb1585"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">viewClade</span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_zoom_in_452-1.png" width="50%" style="display: block; margin: auto;"></div>
</div>
<div id="colapsando-ramos" class="section level3 unnumbered">
<h3>Colapsando ramos<a class="anchor" aria-label="anchor" href="#colapsando-ramos"><i class="fas fa-link"></i></a>
</h3>
<p>No entanto, podemos querer ignorar este ramo e podemos colapsa-lo naquele mesmo nó (nó nº 452) utilizando <code><a href="https://dplyr.tidyverse.org/reference/compute.html">collapse()</a></code>. Esta árvore é definida como <code>p_collapsed</code>.</p>
<div class="sourceCode" id="cb1586"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_collapsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html">collapse</a></span><span class="op">(</span><span class="va">p</span>, node <span class="op">=</span> <span class="fl">452</span><span class="op">)</span></span>
<span><span class="va">p_collapsed</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_collapse_452-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Para maior clareza, quando imprimimos <code>p_collapsed', adicionamos um</code>geom_point2()` (um diamante azul) no nó do ramo colapsado.</p>
<div class="sourceCode" id="cb1587"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_collapsed</span> <span class="op">+</span> </span>
<span><span class="fu">geom_point2</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">(</span><span class="va">node</span> <span class="op">==</span> <span class="fl">452</span><span class="op">)</span><span class="op">)</span>,  <span class="co"># adicionamos um símbolo ao nó colapsado </span></span>
<span>            size <span class="op">=</span> <span class="fl">5</span>,                     <span class="co"># define o tamanho do símbolo  </span></span>
<span>            shape <span class="op">=</span> <span class="fl">23</span>,                   <span class="co"># define a forma do símbolo  </span></span>
<span>            fill <span class="op">=</span> <span class="st">"steelblue"</span><span class="op">)</span>           <span class="co"># define a cor do símbolo </span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1230-1.png" width="672"></div>
</div>
<div id="subdividindo-uma-árvore" class="section level3 unnumbered">
<h3>Subdividindo uma árvore<a class="anchor" aria-label="anchor" href="#subdividindo-uma-%C3%A1rvore"><i class="fas fa-link"></i></a>
</h3>
<p>Se quisermos fazer uma mudança mais permanente e criar uma árvore nova e reduzida para trabalhar com ela, podemos separar parte dela com <code>tree_subset()</code>. Então você pode salvá-la como novo arquivo newick tree ou arquivo .txt.</p>
<p>Primeiro, inspecionamos os nós de árvore e os rótulos das pontas a fim de decidir como subdividi-la.</p>
<div class="sourceCode" id="cb1588"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span></span>
<span>  <span class="va">tree</span>,</span>
<span>  branch.length <span class="op">=</span> <span class="st">'none'</span>,</span>
<span>  layout <span class="op">=</span> <span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>               <span class="co"># adicionamos os dados das amostras usando o operdor  %&lt;+% </span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">+</span>                                <span class="co"># rotula as pontas dos ramos com o nome da amostra </span></span>
<span>  <span class="fu">geom_text2</span><span class="op">(</span></span>
<span>    mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="op">!</span><span class="va">isTip</span>, label <span class="op">=</span> <span class="va">node</span><span class="op">)</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"darkred"</span><span class="op">)</span> <span class="op">+</span>                                <span class="co"># rotula todos os nós na árvore </span></span>
<span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>   legend.position <span class="op">=</span> <span class="st">"none"</span>,                            <span class="co"># remove a legenda completamente</span></span>
<span>   axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>   axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>   plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_subsetting-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Agora, digamos que decidimos dividir a árvore no nó 528 (manter apenas as pontas dentro deste ramo após o nó 528) e a salvamos como um novo objeto <code>sub_tree1</code>:</p>
<div class="sourceCode" id="cb1589"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sub_tree1</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span></span>
<span>  <span class="va">tree</span>,</span>
<span>  node <span class="op">=</span> <span class="fl">528</span><span class="op">)</span>                                            <span class="co"># subdividimos a árvore no nó 528 </span></span></code></pre></div>
<p>Vamos dar uma olhada na árvore do subconjunto 1 (subset tree 1):</p>
<div class="sourceCode" id="cb1590"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Subset tree 1"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1232-1.png" width="672"></div>
<p>Você também pode dividir com base em uma amostra específica, especificando quantos nós “para trás” você deseja incluir. Vamos subdividir a mesma parte da árvore com base em uma amostra, neste caso S17BD07692, retrocedendo 9 nós e a salvamos como um novo objeto <code>sub_tree2</code>:</p>
<div class="sourceCode" id="cb1591"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sub_tree2</span> <span class="op">&lt;-</span> <span class="fu">tree_subset</span><span class="op">(</span></span>
<span>  <span class="va">tree</span>,</span>
<span>  <span class="st">"S17BD07692"</span>,</span>
<span>  levels_back <span class="op">=</span> <span class="fl">9</span><span class="op">)</span> <span class="co">#  levels_back define quantos nós você quer retroceder da ponta da amostra  </span></span></code></pre></div>
<p>Vamos dar uma olhada na árvore do subconjunto 2:</p>
<div class="sourceCode" id="cb1592"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span>  <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Subset tree 2"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1234-1.png" width="672"></div>
<p>Você também pode salvar sua nova árvore como um tipo Newick ou mesmo um arquivo de texto utilizando a função <code>write.tree()</code> do pacote <strong>ape</strong>:</p>
<div class="sourceCode" id="cb1593"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># salvar no formato .nwk </span></span>
<span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.nwk'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># salvar no formato .txt </span></span>
<span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/write.tree.html">write.tree</a></span><span class="op">(</span><span class="va">sub_tree2</span>, file<span class="op">=</span><span class="st">'data/phylo/Shigella_subtree_2.txt'</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="rotacionando-nós-em-uma-árvore" class="section level3 unnumbered">
<h3>Rotacionando nós em uma árvore<a class="anchor" aria-label="anchor" href="#rotacionando-n%C3%B3s-em-uma-%C3%A1rvore"><i class="fas fa-link"></i></a>
</h3>
<p>Como mencionado anteriormente, não podemos mudar a ordem das pontas ou nós na árvore, pois isto se baseia em sua relação genética e não está sujeito a manipulação visual. Mas podemos girar ramos em torno dos nós se isso facilitar nossa visualização.</p>
<p>Primeiro, traçamos nossa nova árvore “sub_tree2” com etiquetas de nós para escolher o nó que queremos manipular e armazená-lo como um objeto do tipo gráfico ggtree <code>p</code>.</p>
<div class="sourceCode" id="cb1594"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">+</span>  </span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_text2</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="op">!</span><span class="va">isTip</span>, label<span class="op">=</span><span class="va">node</span><span class="op">)</span>, <span class="co"># rotula todos os nós em uma árvore</span></span>
<span>             size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>             color <span class="op">=</span> <span class="st">"darkred"</span>, </span>
<span>             hjust <span class="op">=</span> <span class="fl">1</span>, </span>
<span>             vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span><span class="va">p</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_1-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Podemos então manipular os nós aplicando <strong>ggtree::rotate()</strong> ou <strong>ggtree::flip()</strong>:
Nota: para ilustrar quais nós estamos manipulando, aplicamos primeiro a função <strong>geom_hilight()</strong> de <strong>ggtree</strong> para destacar as amostras nos nós em que estamos interessados e armazenar esse objeto ggtree em um novo objeto <code>p1</code>.</p>
<div class="sourceCode" id="cb1595"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu">geom_hilight</span><span class="op">(</span>  <span class="co"># destaca o nó 39 em azul, "extend =" nos permite definir a cor do bloco </span></span>
<span>  node <span class="op">=</span> <span class="fl">39</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"steelblue"</span>,</span>
<span>  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>  </span>
<span><span class="fu">geom_hilight</span><span class="op">(</span>            <span class="co"># destaca o nó 37 em amarelo </span></span>
<span>  node <span class="op">=</span> <span class="fl">37</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"yellow"</span>,</span>
<span>  extend <span class="op">=</span> <span class="fl">0.0017</span><span class="op">)</span> <span class="op">+</span>               </span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Árvore Original"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p1</span> <span class="co"># printa</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_rotating_2-1.png" width="50%" style="display: block; margin: auto;"></div>
<p>Agora podemos girar o nó 37 no objeto <code>p1</code> para que as amostras no nó 38 se movam para o topo. Armazenamos a árvore rotacionada em um novo objeto <code>p2</code>.</p>
<div class="sourceCode" id="cb1596"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/rotate.html">rotate</a></span><span class="op">(</span><span class="va">p1</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span> </span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Nó 37 rotacionado"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p2</span>   <span class="co"># printa</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1235-1.png" width="672"></div>
<p>Ou podemos usar a função <code>flip()</code> para rotacuonar o nó 36 do objeto <code>p1</code> e mudar o nó 37 para o topo e o nó 39 para o fundo. Armazenamos a árvore invertida em um novo objeto <code>p3</code>.</p>
<div class="sourceCode" id="cb1597"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/flip.html">flip</a></span><span class="op">(</span><span class="va">p1</span>, <span class="fl">39</span>, <span class="fl">37</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Nó 36 rotacionado"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p3</span>   <span class="co"># printa</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/unnamed-chunk-1236-1.png" width="672"></div>
</div>
<div id="examplo-de-sub-árvore-com-informações-das-amostras" class="section level3 unnumbered">
<h3>Examplo de sub-árvore com informações das amostras<a class="anchor" aria-label="anchor" href="#examplo-de-sub-%C3%A1rvore-com-informa%C3%A7%C3%B5es-das-amostras"><i class="fas fa-link"></i></a>
</h3>
<p>Digamos que estamos investigando o conjunto de casos com expansão clonal que ocorreram em 2017 e 2018 no nó 39 de nossa sub-árvore. Acrescentamos o ano de isolamento da linhagem, bem como o histórico de viagem e a cor por país para ver a origem de outras linhagens intimamente relacionadas:</p>
<div class="sourceCode" id="cb1598"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span>     <span class="co"># Usamos o operador %&lt;+% para juntar com sample_data</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># rotula as pontas dos ramos com  os nomes das amotras </span></span>
<span>    size <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>    offset <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    <span class="co">#align = TRUE</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span>                       <span class="co"># configura os limite do eixo x </span></span>
<span>  <span class="fu">geom_tippoint</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">Country</span><span class="op">)</span>,     <span class="co"># colore a ponta de acordo com o continente  </span></span>
<span>                size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span><span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Country"</span>, </span>
<span>    palette <span class="op">=</span> <span class="st">"Set1"</span>, </span>
<span>    na.value <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co">#  adiciona o ano de isolamento como um rótulo de texto nas pontas a</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">Year</span><span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">'blue'</span>,</span>
<span>    offset <span class="op">=</span> <span class="fl">0.0045</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    linetype <span class="op">=</span> <span class="st">"blank"</span> ,</span>
<span>    geom <span class="op">=</span> <span class="st">"text"</span>,</span>
<span>    <span class="co">#align = TRUE</span></span>
<span>    <span class="op">)</span><span class="op">+</span> </span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>                          <span class="co"># adiciona histórico de viagem nas pontas em vermelho </span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">Travel_history</span><span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>    offset <span class="op">=</span> <span class="fl">0.006</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    linetype <span class="op">=</span> <span class="st">"blank"</span>,</span>
<span>    geom <span class="op">=</span> <span class="st">"text"</span>,</span>
<span>    <span class="co">#align = TRUE</span></span>
<span>    <span class="op">)</span><span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Árvore filogenética de de estirpes de Belgian S. sonnei com histórico de viagens"</span><span class="op">)</span><span class="op">+</span>  <span class="co"># adiciona título </span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"distância genética (0.001 = 4 nucleotídeos de diferença)"</span><span class="op">)</span><span class="op">+</span>   <span class="co"># adiciona legenda do eixo x</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_inspect_subset_example-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Nossa observação aponta para um evento de importação de cepas da Ásia, que então circularam na Bélgica ao longo dos anos e parecem ter causado nosso último surto.</p>
<!-- ======================================================= -->
</div>
</div>
<div id="árvores-mais-complexas-adição-de-heatmaps-de-dados-de-amostra" class="section level2 unnumbered">
<h2>Árvores mais complexas: adição de heatmaps de dados de amostra<a class="anchor" aria-label="anchor" href="#%C3%A1rvores-mais-complexas-adi%C3%A7%C3%A3o-de-heatmaps-de-dados-de-amostra"><i class="fas fa-link"></i></a>
</h2>
<p>Podemos acrescentar informações mais complexas, como a presença categórica de genes de resistência antimicrobiana e valores numéricos para resistência realmente medida a antimicrobianos na forma de um mapa térmico usando a função <strong>ggtree::gheatmap()</strong>.</p>
<p>Primeiro precisamos traçar nossa árvore (esta pode ser linear ou circular) e armazená-la em um novo objeto de traçado ggtree <code>p</code>: Utilizaremos a sub-árvore da parte 3).</p>
<div class="sourceCode" id="cb1599"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span>, branch.length<span class="op">=</span><span class="st">'none'</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span> </span>
<span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>   legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span></span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap-1.png" width="60%" style="display: block; margin: auto;"></div>
<p>Em segundo lugar, preparamos nossos dados. Para visualizar as diferentes variáveis com novos esquemas de cores, nós subdividimos nosso dataframe para a variável desejada. É importante adicionar o <code>Sample_ID</code> como nomes das linhas, caso contrário, ele não pode fazer a correspondência dos dados com a árvore <code>tip.labels</code>:</p>
<p>Em nosso exemplo, queremos analisar o gênero e as mutações que poderiam conferir resistência à Ciprofloxacina, um importante antibiótico de primeira linha utilizado para tratar as infecções por Shigella.</p>
<p>Criamos um dataframe para gênero:</p>
<div class="sourceCode" id="cb1600"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"gender"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gender"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gender</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></span></code></pre></div>
<p>Criamos um dataframe para mutações no gene gyrA, que confere resistência à Ciprofloxacina:</p>
<div class="sourceCode" id="cb1601"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cipR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"cipR"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gyrA_mutations"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cipR</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></span></code></pre></div>
<p>Criamos um dataframe para a concentração inibitória mínima medida (MIC) de Ciprofloxacina do laboratório:</p>
<div class="sourceCode" id="cb1602"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MIC_Cip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="st">"mic_cip"</span> <span class="op">=</span> <span class="va">sample_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MIC_CIP"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">MIC_Cip</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_data</span><span class="op">$</span><span class="va">Sample_ID</span></span></code></pre></div>
<p>Criamos um primeiro gráfico adicionando um heatmap binário para gênero à árvore filogenética e armazenando-o em um novo objeto do tipo gráfico ggtree <code>h1</code>:</p>
<div class="sourceCode" id="cb1603"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>,                                 <span class="co"># adicionamos uma camada "heatmap" de gênero ao gráfico</span></span>
<span>                offset <span class="op">=</span> <span class="fl">10</span>,                               <span class="co"># "offset" muda o heatmap para a direita,</span></span>
<span>                width <span class="op">=</span> <span class="fl">0.10</span>,                              <span class="co"># "width" define ta largura da coluna do heatmap,</span></span>
<span>                color <span class="op">=</span> <span class="cn">NULL</span>,                              <span class="co"># "color"  define a borda da coluna do  heatmap </span></span>
<span>         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>                               <span class="co"># esconde a coluna de nomes do heatmap</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,                       <span class="co"># define o esquema de cores e a legenda do gênero </span></span>
<span>                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,</span>
<span>                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,</span>
<span>                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1605"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h1</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_gender-1.png" width="70%" style="display: block; margin: auto;"></div>
<p>Depois adicionamos informações sobre mutações no gene gyrA, que conferem resistência ao Ciprofloxacina:</p>
<p>Nota: A presença de mutações cromossômicas pontuais nos dados do WGS foi previamente determinada usando a ferramenta PointFinder desenvolvida por Zankari et al. (ver referência na seção de referências adicionais)</p>
<p>Primeiro, atribuímos um novo esquema de cores ao nosso objeto do tipo gráfico existente em<code>h1</code> e o armazenamos em um objeto agora <code>h2</code>. Isto nos permite definir e mudar as cores para nossa segunda variável no heatmap.</p>
<div class="sourceCode" id="cb1606"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p>Depois adicionamos a segunda camada do heatmap ao <code>h2</code> e armazenamos os gráficos combinados em um novo objeto <code>h3</code>:</p>
<div class="sourceCode" id="cb1607"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,  <span class="co"># adiciona uma segunda linha de heatmap descrevendo a mutação de resistência da Ciprofloxacina</span></span>
<span>               offset <span class="op">=</span> <span class="fl">12</span>, </span>
<span>               width <span class="op">=</span> <span class="fl">0.10</span>, </span>
<span>               colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Mutações que conferem \n resistência a Ciprofloxacina"</span>,</span>
<span>                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,</span>
<span>                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,</span>
<span>                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1609"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h3</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_genes-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Repetimos o processo acima, primeiro adicionando uma nova camada de escala de cor ao nosso objeto existente <code>h3</code>, e depois adicionando os dados contínuos sobre a concentração inibitória mínima (MIC) de Ciprofloxacina para cada estirpe ao objeto resultante <code>h4</code> para produzir o objeto final <code>h5</code>:</p>
<div class="sourceCode" id="cb1610"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Primeiro adicionamos o novo esquema de cores</span></span>
<span><span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># e então combinamos os dois em um novo gráfico:</span></span>
<span><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>,  </span>
<span>               offset <span class="op">=</span> <span class="fl">14</span>, </span>
<span>               width <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_fill_continuous</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC para Ciprofloxacina"</span>,  <span class="co"># definimos uma cor em gradiente para a variável contínua MIC </span></span>
<span>                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.50</span>, <span class="fl">1.00</span><span class="op">)</span>,</span>
<span>                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_colourbar.html">guide_colourbar</a></span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1612"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_cip_MIC-1.png" width="90%" style="display: block; margin: auto;"></div>
<p>Podemos fazer o mesmo exercício para uma árvore linear:</p>
<div class="sourceCode" id="cb1613"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">sub_tree2</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">sample_data</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tiplab</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="co"># rotupa as pontas</span></span>
<span>  <span class="fu">theme_tree2</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"distância gentética (0.001 = 4 nucleotídeos de diferença)"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.015</span><span class="op">)</span><span class="op">+</span></span>
<span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>      axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, </span>
<span>                                face <span class="op">=</span> <span class="st">"bold"</span>,</span>
<span>                                hjust <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                                vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_1-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Primeiro, adicionamos o gênero:</p>
<div class="sourceCode" id="cb1614"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h1</span> <span class="op">&lt;-</span>  <span class="fu">gheatmap</span><span class="op">(</span><span class="va">p</span>, <span class="va">gender</span>, </span>
<span>                offset <span class="op">=</span> <span class="fl">0.003</span>,</span>
<span>                width <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>                color<span class="op">=</span><span class="st">"black"</span>, </span>
<span>         colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Gender"</span>,</span>
<span>                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#00d1b1"</span>, <span class="st">"purple"</span><span class="op">)</span>,</span>
<span>                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span>,</span>
<span>                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Male"</span>, <span class="st">"Female"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1616"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h1</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_2-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Depois adicionamos as mutações de resistência Ciprofloxacina depois de adicionar outra camada de esquema de cores:</p>
<div class="sourceCode" id="cb1617"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h2</span> <span class="op">&lt;-</span> <span class="va">h1</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">h3</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h2</span>, <span class="va">cipR</span>,   </span>
<span>               offset <span class="op">=</span> <span class="fl">0.004</span>, </span>
<span>               width <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Mutações que conferem \n resistência a Ciprofloxacina"</span>,</span>
<span>                    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#fe9698"</span>,<span class="st">"#ea0c92"</span><span class="op">)</span>,</span>
<span>                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA D87Y"</span>, <span class="st">"gyrA S83L"</span><span class="op">)</span>,</span>
<span>                    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"gyrA d87y"</span>, <span class="st">"gyrA s83l"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"vertical"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">2</span>,byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1619"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="va">h3</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_3-1.png" width="80%" style="display: block; margin: auto;"></div>
<p>Em seguida, adicionamos a concentração inibitória mínima determinada pelo laboratório (MIC):</p>
<div class="sourceCode" id="cb1620"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h4</span> <span class="op">&lt;-</span> <span class="va">h3</span> <span class="op">+</span> <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">gheatmap</span><span class="op">(</span><span class="va">h4</span>, <span class="va">MIC_Cip</span>, </span>
<span>               offset <span class="op">=</span> <span class="fl">0.005</span>,  </span>
<span>               width <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>               color <span class="op">=</span> <span class="st">"black"</span>, </span>
<span>                colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_fill_continuous</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"MIC para Ciprofloxacina"</span>,</span>
<span>                      low <span class="op">=</span> <span class="st">"yellow"</span>, high <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>                      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.50</span>,<span class="fl">1.00</span><span class="op">)</span>,</span>
<span>                      na.value <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span><span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_colourbar.html">guide_colourbar</a></span><span class="op">(</span>barwidth <span class="op">=</span> <span class="fl">5</span>, barheight <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>        legend.box <span class="op">=</span> <span class="st">"horizontal"</span>, legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scale for y is already present.
## Adding another scale for y, which will replace the existing scale.
## Scale for fill is already present.
## Adding another scale for fill, which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb1622"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5</span></span></code></pre></div>
<div class="inline-figure"><img src="_main_files/figure-html/phylogenetic_trees_sampledata_heatmap_linear_4-1.png" width="80%" style="display: block; margin: auto;"></div>
<!-- ======================================================= -->
</div>
<div id="resources" class="section level2" number="36.5">
<h2>
<span class="header-section-number">36.5</span> Resources<a class="anchor" aria-label="anchor" href="#resources"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#" class="uri">http://hydrodictyon.eeb.uconn.edu/eebedia/index.php/Ggtree#</a> Clade_Colors
<a href="https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.2/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a>
<a href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html" class="uri">https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html</a>
<a href="https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html" class="uri">https://bioconductor.riken.jp/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeManipulation.html</a></p>
<p>Ea Zankari, Rosa Allesøe, Katrine G Joensen, Lina M Cavaco, Ole Lund, Frank M Aarestrup, PointFinder: a novel web tool for WGS-based detection of antimicrobial resistance associated with chromosomal point mutations in bacterial pathogens, Journal of Antimicrobial Chemotherapy, Volume 72, Issue 10, October 2017, Pages 2764–2768, <a href="https://doi.org/10.1093/jac/dkx217" class="uri">https://doi.org/10.1093/jac/dkx217</a></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="transmission-chains.html"><span class="header-section-number">35</span> Cadeias de Transmissão</a></div>
<div class="next"><a href="interactive-plots.html"><span class="header-section-number">37</span> Gráficos interativos</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#phylogenetic-trees"><span class="header-section-number">36</span> Árvores filogenéticas</a></li>
<li><a class="nav-link" href="#vis%C3%A3o-geral-5"><span class="header-section-number">36.1</span> Visão Geral</a></li>
<li>
<a class="nav-link" href="#prepara%C3%A7%C3%A3o-28"><span class="header-section-number">36.2</span> Preparação</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#carregar-pacotes-11">Carregar pacotes</a></li>
<li><a class="nav-link" href="#importar-dados-4">Importar dados</a></li>
<li><a class="nav-link" href="#limpar-e-inspecionar">Limpar e inspecionar</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#visualiza%C3%A7%C3%A3o-simples-da-%C3%A1rvore"><span class="header-section-number">36.3</span> Visualização simples da árvore</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#diferente-layouts-para-a-%C3%A1rvore">Diferente layouts para a árvore</a></li>
<li><a class="nav-link" href="#%C3%A1rvore-simples-com-dados-de-amostras">Árvore simples com dados de amostras</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#manipula%C3%A7%C3%A3o-da-%C3%A1rvore"><span class="header-section-number">36.4</span> Manipulação da árvore</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#aumentar-o-zoom">Aumentar o zoom</a></li>
<li><a class="nav-link" href="#colapsando-ramos">Colapsando ramos</a></li>
<li><a class="nav-link" href="#subdividindo-uma-%C3%A1rvore">Subdividindo uma árvore</a></li>
<li><a class="nav-link" href="#rotacionando-n%C3%B3s-em-uma-%C3%A1rvore">Rotacionando nós em uma árvore</a></li>
<li><a class="nav-link" href="#examplo-de-sub-%C3%A1rvore-com-informa%C3%A7%C3%B5es-das-amostras">Examplo de sub-árvore com informações das amostras</a></li>
</ul>
</li>
<li><a class="nav-link" href="#%C3%A1rvores-mais-complexas-adi%C3%A7%C3%A3o-de-heatmaps-de-dados-de-amostra">Árvores mais complexas: adição de heatmaps de dados de amostra</a></li>
<li><a class="nav-link" href="#resources"><span class="header-section-number">36.5</span> Resources</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Manual de R para Epidemiologistas</strong>" was written by the handbook team. It was last built on 2023-10-22.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
